### PR TITLE
Add regression tests for bug #2228

### PR DESCRIPTION
**Pull request by rabbitonweb**
Fixes https://github.com/quasar-analytics/quasar/issues/2228

### CODE REVIEW

File: it/testing.conf.example [hunk 1]
```diff
@@ -4,7 +4,7 @@
 #mongodb_3_2       ="mongodb://<host>:<port>"
 #mongodb_read_only ="mongodb://<host>:<port>"
 #spark_local       ="<path_to_folder>"
-#spark_hdfs        ="spark://<spark_host>:<spark_port>?hdfsUrl=hdfs://<hdfs_host>:<hdfs_port>&rootPath=<root_path>"
+# spark_hdfs        = "spark://hideout:7077?hdfsUrl=hdfs://serv1.demo.slamdata.com:9000&rootPath=/"
```
Comment (djspiewak) [inline it/testing.conf.example]: Spurious whitespace

File: it/src/main/resources/tests/bug2228.test [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "bug2228",
+  "data": ["customers.data", "orders.data", "ordered_items.data", "inventory_items.data"],
+  "backends": {
+    "couchbase":         "ignoreFieldOrder",
+    "marklogic_json":    "ignoreFieldOrder",
+    "mimir":             "pending"
```
Comment (djspiewak) [inline it/src/main/resources/tests/bug2228.test]: Can you add this to #2542 with a quick note as to what happens?

**Comment by djspiewak** [comment]
Needs version label.  Also a `skip` for Couchbase.

File: it/testing.conf.example (hunk 1 above)
Comment (alissapajer) [reply it/testing.conf.example]: Do we want to be checking this line in, showing demo server access as an example?

File: it/src/main/resources/tests/bug2228.test (hunk 2 above)
Comment (alissapajer) [reply it/src/main/resources/tests/bug2228.test]: `mimir` should be `pendingIgnoreFieldOrder`.

File: it/testing.conf.example (hunk 1 above)
Comment (djspiewak) [reply it/testing.conf.example]: Uh… good call. @rabbitonweb you need to amend this commit asap to remove this data.  Don't add a commit; `amend` and `push -f`.

**Comment by djspiewak** [comment]
@rabbitonweb @damonLL We've scrubbed the HDFS cluster information from the commits.  However, the damage is already done.  That cluster needs to be authenticated (it should have been authenticated all along, though that wouldn't have saved us here since the credentials would have been committed).  Any data in that cluster should now be considered public, and the cluster itself can be safely considered open to the public until it is secured.

**Comment by rabbitonweb** [comment]
@djspiewak I don't quite follow your comment. Do you mean the data used for tests is private? Or do you mean we should have secure HDFS?

**Comment by rabbitonweb** [comment]
version label added, ready for merge

**Comment by djspiewak** [comment]
@rabbitonweb I mean we shouldn't be exposing services to the open internet without security.  Whether or not the data contained within HDFS is private, HDFS *itself* is not intended to be exposed in that way.  The service is designed to run behind a firewall, exposed only to application servers (and in some environments, only *certain* application servers).  The only "security" we have on our HDFS cluster is the fact that no one knew we had it.  That has now been stripped away.

The right thing to do is firewall off HDFS from everything except the demo server.  Remote access (for dev or demo purposes) can be conveniently and securely achieved via SSH tunneling.

**Comment by alissapajer** [comment]
@rabbitonweb there are conflicts to resolve.

**Comment by alissapajer** [comment]
Huh, this now appears to be merged...

**Pull request merged**